### PR TITLE
th-126: The Password field content on the Sign up both Business and Customer and Sign in pages is hidden by asterisks after the Show Hide password button was clicked

### PR DESCRIPTION
**Pull request by albsbz**
Add show/hide password behaviour
pw1
pw2

### CODE REVIEW

File: frontend/src/libs/components/input/input.tsx [hunk 1]
```diff
@@ -42,19 +46,29 @@ const Input = <T extends FieldValues>({
...
           {...field}
-          type={type}
+          type={showPassword ? 'text' : type}
           placeholder={placeholder}
           className={getValidClassNames(...inputStyles)}
           disabled={isDisabled}
         />
         {type === 'password' && (
-          <span className={styles.passwordEye}>&#128065;</span>
+          <input
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/input/input.tsx]: can we add it as button, not as input type="button"? 
also what about changing eye icon when turning on/off? now it's not working this way

Comment (albsbz) [reply frontend/src/libs/components/input/input.tsx]: fixed

File: frontend/src/libs/components/input/input.tsx [hunk 2]
```diff
@@ -30,7 +36,7 @@ const Input = <T extends FieldValues>({
   isDisabled,
 }: Properties<T>): JSX.Element => {
   const { field } = useFormController({ name, control });
-
+  const [showPassword, setShowPassword] = useState(false);
```
Comment (h0wter) [inline frontend/src/libs/components/input/input.tsx:39]: It is better to call boolean using be/have verbs. For example: `isPasswordShown`.

Comment (h0wter) [inline frontend/src/libs/components/input/input.tsx]: It is better to call boolean using be/have verbs. For example: `isPasswordShown`.

File: frontend/src/libs/components/input/styles.module.scss [hunk 3]
```diff
@@ -86,7 +86,10 @@ $error-message-height: calc($error-font-size * $line-height-coefficient);
 .passwordEye {
   position: absolute;
   top: 8px;
-  right: 15px;
+  right: 10px;
+  color: $grey-light;
+  background-color: $white;
```
Comment (h0wter) [inline frontend/src/libs/components/input/styles.module.scss:91]: Maybe we should make different state icons have different colors?

Comment (albsbz) [reply frontend/src/libs/components/input/styles.module.scss:91]: You mean on hover, or depending if password showed or not?

Comment (h0wter) [reply frontend/src/libs/components/input/styles.module.scss:91]: I meant depending on the state, shown or not.

**Review by DmytroVoronkov — APPROVED** [review]

File: frontend/src/libs/components/input/styles.module.scss (hunk 3 above)
Comment (albsbz) [reply frontend/src/libs/components/input/styles.module.scss:91]: Added color changing

**Review by h0wter — APPROVED** [review]

File: frontend/src/libs/components/input/input.tsx (hunk 2 above)
Comment (albsbz) [reply frontend/src/libs/components/input/input.tsx]: fixed

File: frontend/src/libs/components/input/input.tsx [hunk 4]
```diff
@@ -30,7 +36,7 @@ const Input = <T extends FieldValues>({
   isDisabled,
 }: Properties<T>): JSX.Element => {
   const { field } = useFormController({ name, control });
-
+  const [isShowPassword, setIsShowPassword] = useState(false);
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/input/input.tsx]:
```suggestion
  const [isPasswordShown, setIsPasswordShown] = useState(false);
```

File: frontend/src/libs/components/input/input.tsx [hunk 5]
```diff
@@ -42,19 +48,36 @@ const Input = <T extends FieldValues>({
...
         />
         {type === 'password' && (
-          <span className={styles.passwordEye}>&#128065;</span>
+          <button
+            className={getValidClassNames(
+              styles.passwordEye,
+              isShowPassword && styles.passwordEyeLight,
+            )}
+            onClick={toggleShowPassword}
+            tabIndex={-1}
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/input/input.tsx:86]: do we need tabIndex with button here?

File: frontend/src/libs/components/input/input.tsx [hunk 6]
```diff
@@ -42,19 +48,36 @@ const Input = <T extends FieldValues>({
     hasError && styles.error,
   ];
 
+  const toggleShowPassword = useCallback(
+    (event: React.MouseEvent<HTMLElement>): void => {
+      event.preventDefault();
```
Comment (kalinkaaaa14) [inline frontend/src/libs/components/input/input.tsx:59]: can we remove prevent default? for which purpose it's used here

File: frontend/src/libs/components/input/input.tsx (hunk 5 above)
Comment (albsbz) [reply frontend/src/libs/components/input/input.tsx:86]: If remove it when tabing through form eye symbol will be in focus,  It seems for me as not predictable behavior

File: frontend/src/libs/components/input/input.tsx (hunk 6 above)
Comment (albsbz) [reply frontend/src/libs/components/input/input.tsx:59]: When I remove it click event bubbles up and triggers form as touched, so all field becomes red

File: frontend/src/libs/components/input/input.tsx (hunk 4 above)
Comment (albsbz) [reply frontend/src/libs/components/input/input.tsx]: fixed

**Review by kalinkaaaa14 — DISMISSED** [review]

File: frontend/src/libs/components/input/input.tsx [hunk 7]
```diff
@@ -5,9 +5,15 @@ import {
   type FieldValues,
 } from 'react-hook-form';
 
+import { IconName } from '~/libs/enums/icon-name.enum';
```
Comment (PavloSukhinin) [inline frontend/src/libs/components/input/input.tsx]: add file extension, please

**Review by PavloSukhinin — CHANGES_REQUESTED** [review]

**Review by PavloSukhinin — APPROVED** [review]

**Review by kalinkaaaa14 — APPROVED** [review]

**Review by Druxight — APPROVED** [review]

**Pull request merged**